webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala

```diff
@@ -130,6 +130,30 @@ object SecurityToken {
       _      <- checkCertificateSubject(nodeId, parsed._2)
     } yield ()
   }
+
+  // Use bouncy castle to parse the key and check if it's a public key or a certificate
+  def parseValidate(key: String): Either[InventoryError, SecurityToken] = {
+    PureResult
+      .attempt(
+        new PEMParser(new StringReader(key))
+          .readObject()
+      )
+      .left
+      .map(ex => InventoryError.CryptoEx(s"Key '${key}' cannot be parsed as a public key", ex.cause): InventoryError)
+      .flatMap { obj =>
+        obj match {
+          case _: SubjectPublicKeyInfo  => Right(PublicKey(key))
```

Comment:
we only support certificate now, but it's ok to do it that way - the removing can be an other issue (even in an other version of rudder)